### PR TITLE
M3-4059 [LKE]: Add Tags to LKE Summary Detail Panel

### PR DESCRIPTION
**Pull request by DevDW**
## Description
This adds tags functionality to the LKE Summary Detail panel. Amended the `KubernetesCluster` interface; passed a `handleUpdateTags` prop to KubeSummaryDetail component from KubernetesClusterDetail. 

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

### CODE REVIEW

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx [hunk 1]
```diff
@@ -99,6 +99,10 @@ interface Props {
   endpointLoading: boolean;
   kubeconfigAvailable: boolean;
   kubeconfigError?: string;
+  handleUpdateTags: (
+    clusterID: number,
+    updatedCluster: ExtendedCluster
```
Comment (johnwcallahan) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx:104]: No reason for this function to accept the entire cluster. You only use `tags`, so you can type it has `handleUpdateTags: (clusterID: number, updatedTags: string[])`. That'll simplify a few things.

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx [hunk 2]
```diff
@@ -99,6 +99,10 @@ interface Props {
...
+  ) => Promise<any>;
```
Comment (johnwcallahan) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx:105]: This should be `Promise<KubernetesCluster>`. You can see the typing of `props.updateCluster` in KubernetesClusterDetail:
Screen Shot 2020-04-13 at 8 24 18 AM

**Comment by johnwcallahan** [comment]
@DevDW let's also use the new styles as shown in the design:

- The white "plus" should be a bit smaller
- The "Add a tag" text should be bold and blue
- Use the new "X" icon on each tag
- There is a bit more padding in each tag element

Given these changes, the difficulty of conditionally positioning the "Add" button, and our timeframe, I think it's acceptable to create a new component, copied from the current `<TagsPanel />` component, with these changes. Then when this tags pattern is applied to other entities in the coming months, we can remove the old one.

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx [hunk 3]
```diff
@@ -193,13 +198,9 @@ export const KubeSummaryPanel: React.FunctionComponent<CombinedProps> = props =>
...
 
-  // The function below is a placeholder that will need to be edited once cluster Tags functionality is in place.
-  /* const updateTags = async (tags: string[]) => {
-    const { cluster.id, clusterActions } = props;
-
-    // Send the request (which updates the internal store.)
-    await clusterActions.updateCluster({ cluster.id, tags });
-  }; */
+  const updateTags = async (updatedTags: string[]) => {
+    await handleUpdateTags(cluster.id, { ...cluster, tags: updatedTags });
```
[the remark this answers is not in the document]
Comment (Jskobos) [reply packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx:202]: Once you've made the change to only send `string[]` as the payload, this wrapper is no longer necessary. If you use `handleUpdateTags` directly on line 358 everything will work out the same.

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx [hunk 4]
```diff
@@ -99,6 +99,10 @@ interface Props {
   endpointLoading: boolean;
   kubeconfigAvailable: boolean;
   kubeconfigError?: string;
+  handleUpdateTags: (
```
Comment (Jskobos) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/KubeSummaryPanel.tsx:102]: `handleUpdateTags: (tags: string[]) => Promise<KubernetesCluster>`

File: packages/manager/src/components/TagsPanel/TagsPanelRedesigned.tsx [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+    addButtonText: {
+      color: theme.palette.primary.main,
+      fontWeight: 700
+    },
+    tagsPanelItemWrapper: {
+      marginBottom: theme.spacing(2),
+      position: 'relative'
+    },
+    selectTag: {
+      marginTop: theme.spacing(1),
```
Comment (WilkinsKa1) [inline packages/manager/src/components/TagsPanel/TagsPanelRedesigned.tsx:109]: Let's remove the marginTop, add a minWidth: 275, and textAlign: left to override the parent's alignment to address the issues John called out.

File: packages/manager/src/components/TagsPanel/TagsPanelRedesigned.tsx [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+          />
+        ) : (
+          <div
+            className={classNames({
+              [classes.addButtonWrapper]: true,
+              [classes.hasError]: tagError
+            })}
+          >
+            <IconTextLink
+              text="Add a Tag"
```
Comment (tiffwong) [inline packages/manager/src/components/TagsPanel/TagsPanelRedesigned.tsx:390]: Should have a lowercase "t" => "Add a tag"

**Pull request merged**